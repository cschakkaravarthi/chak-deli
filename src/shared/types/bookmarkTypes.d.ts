import Bookmark from '../models/Bookmark.model';
export declare const LOAD_BOOKMARKS = "LOAD_BOOKMARKS";
export declare const FETCH_BOOKMARKS = "FETCH_BOOKMARKS";
export declare const SET_BOOKMARK = "SET_BOOKMARK";
export declare const SAVE_BOOKMARK = "SAVE_BOOKMARK";
export declare const DELETE_BOOKMARK = "DELETE_BOOKMARK";
export declare const REMOVE_BOOKMARK = "REMOVE_BOOKMARK";
export interface LoadBookmarksAction {
    type: typeof LOAD_BOOKMARKS;
    bookmarks: Bookmark[];
}
export interface SetBookmarkAction {
    type: typeof SET_BOOKMARK;
    success: boolean;
}
export interface DeleteBookmarkAction {
    type: typeof DELETE_BOOKMARK;
    success: boolean;
}
export interface BookmarkState {
    bookmarks: Bookmark[];
    success: boolean;
}
//# sourceMappingURL=bookmarkTypes.d.ts.map