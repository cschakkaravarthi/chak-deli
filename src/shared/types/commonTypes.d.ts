export declare const TOGGLE_GLOBAL_TOAST = "TOGGLE_GLOBAL_TOAST";
export interface ToggleGlobalToastAction {
    type: typeof TOGGLE_GLOBAL_TOAST;
    displayToast: boolean;
    toastMessage: string;
    toastError: boolean;
}
export interface CommonState {
    displayToast: boolean;
    toastMessage: string;
    toastError: boolean;
}
//# sourceMappingURL=commonTypes.d.ts.map