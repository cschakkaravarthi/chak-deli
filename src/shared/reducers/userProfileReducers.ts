import { Reducer } from 'redux';
import {
  UserProfileState,
  SET_USER_PROFILE_DETAILS,
  SET_ORG_CHART,
  USER_PROFILE_RESPONSE_ERROR,
  ORG_CHART_RESPONSE_ERROR,
  CLEAR_PROFILE_DATA,
  SET_USER_PROFILE_PICTURE,
  UPDATE_USER_PROFILE_DATA,
  UPDATE_USER_PROFILE_REQUEST,
  UPDATE_USER_PROFILE_ERROR,
  CLEAR_USER_PROFILE_UPDATE_ERROR,
  FETCH_USER_PROFILE,
  FETCH_MY_CONTACTS,
  FETCH_ORG_CHART,
  SET_MY_CONTACTS,
  MY_CONTACTS_RESPONSE_ERROR,
  REMOVE_SINGLE_CONTACT,
  INITIATE_UPLOAD_PROFILE_PIC,
  UPLOAD_PROFILE_PIC_SUCCESS,
  UPLOAD_PROFILE_PIC_ERROR,
  REMOVE_PROFILE_PIC,
  COMPLETE_REMOVE_PROFILE_PIC
} from '../types/userProfileTypes';
import { Users } from 'umgc_ui_library/lib/seeds/OrgChart.model';
import { UserProfileModel, UserProfilePicture, ContactList } from '../models/UserInfo.model';

export const initialState = {
  userProfileDetails: {} as UserProfileModel,
  userOrgChart: {} as Users,
  profileDataError: false as boolean,
  chartDataError: false as boolean,
  userProfilePicture: {} as UserProfilePicture,
  isUserProfileUpdate: true as boolean,
  isFetchUserProfile: true as boolean,
  userProfileUpdateErrorMessage: '',
  isUserProfileUpdateError: false,
  isFetchingMyContacts: true,
  fetchingMyContactsError: false,
  userContactList: {} as ContactList[],
  removeContactSucess: false,
  isUploading: false,
  uploadSuccess: false,
  uploadError: false,
  removeProfilePicSucess: false,
  isFetchingOrgChart: true
};

const userProfileReducers: Reducer<UserProfileState, any> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case SET_USER_PROFILE_DETAILS:
      return {
        ...state,
        userProfileDetails: action.userProfile,
        profileDataError: false,
        isFetchUserProfile: false
      };
    case FETCH_USER_PROFILE:
      return {
        ...state,
        isFetchUserProfile: action.isFetchUserProfile,
        uploadSuccess: action.uploadSuccess
      };
    case SET_USER_PROFILE_PICTURE:
      return {
        ...state,
        userProfilePicture: action.userProfilePicture,
        profileDataError: false
      };
    case FETCH_ORG_CHART:
      return {
        ...state,
        isFetchingOrgChart: action.isFetchingOrgChart
      };
    case SET_ORG_CHART:
      return {
        ...state,
        userOrgChart: action.orgChartusers,
        chartDataError: action.chartDataError,
        isFetchingOrgChart: action.isFetchingOrgChart
      };
    case USER_PROFILE_RESPONSE_ERROR:
      return {
        ...state,
        profileDataError: true,
        isFetchUserProfile: false
      };
    case ORG_CHART_RESPONSE_ERROR:
      return {
        ...state,
        chartDataError: true
      };
    case CLEAR_PROFILE_DATA:
      return {
        ...state,
        userProfileDetails: { title: '' },
        userOrgChart: {},
        profileDataError: false,
        chartDataError: false
      };
    case UPDATE_USER_PROFILE_DATA:
      return {
        ...state,
        isUserProfileUpdate: action.isUserProfileUpdate,
        userProfileUpdateErrorMessage: action.userProfileUpdateErrorMessage,
        isUserProfileUpdateError: action.isUserProfileUpdateError
      };
    case UPDATE_USER_PROFILE_REQUEST:
      return {
        ...state,
        isUserProfileUpdate: action.isUserProfileUpdate
      };
    case UPDATE_USER_PROFILE_ERROR:
      return {
        ...state,
        isUserProfileUpdate: action.isUserProfileUpdate
      };
    case CLEAR_USER_PROFILE_UPDATE_ERROR:
      return {
        ...state,
        isUserProfileUpdateError: action.isUserProfileUpdateError,
        userProfileUpdateErrorMessage: action.userProfileUpdateErrorMessage
      };
    case FETCH_MY_CONTACTS:
      return {
        ...state,
        isFetchingMyContacts: action.isFetchingMyContacts
      };
    case SET_MY_CONTACTS:
      return {
        ...state,
        userContactList: action.userContactList,
        isFetchingMyContacts: action.isFetchingMyContacts,
        removeContactSucess: action.removeContactSucess
      };
    case MY_CONTACTS_RESPONSE_ERROR:
      return {
        ...state,
        isFetchingMyContacts: action.isFetchingMyContacts,
        fetchingMyContactsError: action.fetchingMyContactsError
      };
    case REMOVE_SINGLE_CONTACT:
      return {
        ...state,
        removeContactSucess: action.removeContactSucess
      };
    case INITIATE_UPLOAD_PROFILE_PIC:
      return {
        ...state,
        uploadSuccess: action.uploadSuccess,
        isUploading: action.isUploading
      };
    case UPLOAD_PROFILE_PIC_SUCCESS:
      return {
        ...state,
        uploadSuccess: action.uploadSuccess,
        isUploading: action.isUploading
      };
    case UPLOAD_PROFILE_PIC_ERROR:
      return {
        ...state,
        uploadError: action.uploadError
      };
    case REMOVE_PROFILE_PIC:
      return {
        ...state,
        removeProfilePicSucess: action.removeProfilePicSucess
      };
    case COMPLETE_REMOVE_PROFILE_PIC:
      return {
        ...state,
        removeProfilePicSucess: action.remo
      };
    default:
      return state;
  }
};

export default userProfileReducers;
