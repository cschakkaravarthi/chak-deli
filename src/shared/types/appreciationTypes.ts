import { AppreciationCardNotificationModel as AlertModel } from '../models/Notification.model';
import { AppreciationCardTemplate, AppreciationCardsListGrouped } from '../models/AppreciationCards.model';

export const FETCH_APPRECIATION_DATA = 'FETCH_APPRECIATION_DATA';
export const LOAD_APPRECIATION_ALERT = 'LOAD_APPRECIATION_ALERT';
export const LOAD_APPRECIATION_ALERT_ERROR = 'LOAD_APPRECIATION_ALERT_ERROR';
export const CLEAR_APPRECIATION_REDUCER = 'CLEAR_APPRECIATION_REDUCER';
export const LOAD_APPRECIATION_MANAGER_ALERT = 'LOAD_APPRECIATION_MANAGER_ALERT';
export const SET_NEW_CARD_DETAIL = 'SET_NEW_CARD_DETAIL';
export const FETCH_CARD_TEMPLATES = 'FETCH_CARD_TEMPLATES';
export const CLEAR_NEW_CARD_FORM = 'CLEAR_NEW_CARD_FORM';
export const TOGGLE_SUCCESS_SENT_ALERT = 'TOGGLE_SUCCESS_SENT_ALERT';
export const GET_APPRECIATION_CARDS_LIST_GROUPED = 'GET_APPRECIATION_CARDS_LIST_GROUPED';
export const VIEW_APPRECIATION_CARD = 'VIEW_APPRECIATION_CARD';
export const APPRECIATION_CARD_ID = 'APPRECIATION_CARD_ID';
export const ARCHIVE_ALERT_STATE = 'ARCHIVE_ALERT_STATE';
export const LOAD_CARDS_IN_GROUP = 'LOAD_CARDS_IN_GROUP';

export interface AppreciationState {
  cardId: string;
  cardArchiveSuccess: boolean;
  isFetching: boolean;
  appreciationCardAlert?: AlertModel[];
  appreciationCardAlertError?: boolean;
  appreciationManagerCardAlert?: AlertModel[];
  newAppreciationCardForm: NewAppreciationCardForm;
  cardTemplates: AppreciationCardTemplate[];
  showSentSucessAlert: boolean;
  appreciationCardsListGrouped?: AppreciationCardsListGrouped;
  cardToView?: AlertModel;
  displayCardView: boolean;
}

export interface NewAppreciationCardForm {
  senderName?: string;
  templateId?: string;
  cardMessage?: string;
  recipientEmail?: string;
  recipientLastName?: string;
  recipientFirstName?: string;
  recipientSupervisorEmail?: string;
  managerName?: string;
}

export interface ArchiveAlertState {
  cardArchiveSuccess: boolean;
  type: typeof ARCHIVE_ALERT_STATE;
}

export interface SetArchiveAlert {
  cardId: string;
  type: typeof APPRECIATION_CARD_ID;
}

export interface FetchRequest {
  isFetching: boolean;
  type: typeof FETCH_APPRECIATION_DATA;
}

export interface LoadAppreciationAlertAction {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_ALERT;
  appreciationCardAlert: AlertModel[];
}

export interface LoadAppreciationManagerAlertAction {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_MANAGER_ALERT;
  appreciationManagerCardAlert: AlertModel[];
}

export interface AppreciationCardAlertError {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_ALERT_ERROR;
  appreciationCardAlertError?: boolean;
}

export interface ClearAprreciationAction {
  type: typeof CLEAR_APPRECIATION_REDUCER;
}

export interface SetNewCardDetailAction {
  type: typeof SET_NEW_CARD_DETAIL;
  payload: NewAppreciationCardForm;
}

export interface SetDetailGroupAction {
  type: typeof SET_NEW_CARD_DETAIL;
  payload: NewAppreciationCardForm;
}

export interface FetchAppreciationCardTemplatesAction {
  type: typeof FETCH_CARD_TEMPLATES;
  cardTemplates: AppreciationCardTemplate[];
}

export interface ClearNewCardFormAction {
  type: typeof CLEAR_NEW_CARD_FORM;
}

export interface ToggleSentSuccessAlert {
  type: typeof TOGGLE_SUCCESS_SENT_ALERT;
  showSentSucessAlert: boolean;
}

export interface GetAppreciationCardsListGroupedAction {
  type: typeof GET_APPRECIATION_CARDS_LIST_GROUPED;
  appreciationCardsListGrouped: AppreciationCardsListGrouped;
}

export interface ViewAppreciationCardAction {
  type: typeof VIEW_APPRECIATION_CARD;
  displayCardView: boolean;
  cardToView?: AlertModel;
}

export interface AddAppreciationCardsToGroupAction {
  type: typeof LOAD_CARDS_IN_GROUP;
  appreciationCards: AlertModel[];
  cardsType: string;
}
