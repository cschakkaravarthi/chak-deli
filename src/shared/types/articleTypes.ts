import Article from '../models/Article.model';
import Facet, { FacetProperty } from '../models/Facet.model';
import { VariousContentModel } from './contentTypes';

export const LOAD_ARTICLES = 'LOAD_ARTICLES';
export const LOAD_ARTICLE_BY_ID = 'LOAD_ARTICLE_BY_ID';
export const FETCH_MORE_ARTICLES = 'LOAD_MORE_ARTICLES';
export const SET_ARTICLE_FILTER_FACET = 'SET_ARTICLE_FILTER_FACET';
export const CLEAR_ARTICLES_LIST = 'CLEAR_ARTICLES_LIST';
export const LIKE_ARTICLE_TOGGLE = 'LIKE_ARTICLE_TOGGLE';

export interface ArticleState {
  article?: Article;
  isLastPage: boolean;
  articleList: Article[];
  facets: Facet;
  selectedFacet: string;
  pageCount: number;
}

export interface LoadArticlesAction {
  type: typeof LOAD_ARTICLES;
  articleList: Article[];
}

export interface LoadArticleByIdAction {
  type: typeof LOAD_ARTICLE_BY_ID;
  article: VariousContentModel;
}
export interface LoadMoreArticlesAction {
  type: typeof FETCH_MORE_ARTICLES;
  articleList?: Article[];
  facets?: Facet;
}
export interface SetFilterFacetAction {
  type: typeof SET_ARTICLE_FILTER_FACET;
  selectedFacet: FacetProperty;
}

export interface ClearArticlesListAction {
  type: typeof CLEAR_ARTICLES_LIST;
}

export interface LikeArticleToggleAction {
  type: typeof LIKE_ARTICLE_TOGGLE;
  articleId: number;
  likedByUser: boolean;
}
