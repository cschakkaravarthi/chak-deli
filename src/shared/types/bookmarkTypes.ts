import Bookmark from '../models/Bookmark.model';

export const LOAD_BOOKMARKS = 'LOAD_BOOKMARKS';
export const FETCH_BOOKMARKS = 'FETCH_BOOKMARKS';
export const SET_BOOKMARK = 'SET_BOOKMARK';
export const SAVE_BOOKMARK = 'SAVE_BOOKMARK';
export const DELETE_BOOKMARK = 'DELETE_BOOKMARK';
export const REMOVE_BOOKMARK = 'REMOVE_BOOKMARK';

export interface LoadBookmarksAction {
  type: typeof LOAD_BOOKMARKS;
  bookmarks: Bookmark[];
}

export interface SetBookmarkAction {
  type: typeof SET_BOOKMARK;
  success: boolean;
}

export interface DeleteBookmarkAction {
  type: typeof DELETE_BOOKMARK;
  success: boolean;
}

export interface BookmarkState {
  bookmarks: Bookmark[];
  success: boolean;
}
