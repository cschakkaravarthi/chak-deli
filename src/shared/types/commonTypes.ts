import RejectModal from './../models/RejectModal';

export const CURRENT_USER_NAME = 'CURRENT_USER_NAME';
export const TOGGLE_GLOBAL_TOAST = 'TOGGLE_GLOBAL_TOAST';
export const SET_REJECT_MODAL_TEXTAREA = 'SET_REJECT_MODAL_TEXTAREA';
export const SET_REJECT_MODAL_ID_ISSHOW = 'SET_REJECT_MODAL_ID_ISSHOW';

export interface ToggleGlobalToastAction {
  type: typeof TOGGLE_GLOBAL_TOAST;
  displayToast: boolean;
  toastMessage: string;
  toastError: boolean;
}

export type CurrentUser = {
  family_name: string;
  given_name: string;
};

export interface SetCurrentUsername {
  type: typeof CURRENT_USER_NAME;
  currentUsername: CurrentUser;
}

export interface CommonState {
  currentUsername: CurrentUser;
  displayToast: boolean;
  toastMessage: string;
  toastError: boolean;
  idAndIsShow: RejectModal;
  textarea: string;
}

export interface SetRejectModalIdIsShow {
  type: typeof SET_REJECT_MODAL_ID_ISSHOW;
  idAndIsShow: RejectModal;
}

export interface SetRejectModalTextarea {
  type: typeof SET_REJECT_MODAL_TEXTAREA;
  textarea: string;
}
