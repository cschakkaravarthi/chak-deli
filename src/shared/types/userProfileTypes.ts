import { UserProfileModel, UserProfilePicture, ContactList } from '../models/UserInfo.model';
import { Users } from '../models/OrgChart.model';

export const FETCH_USER_PROFILE = 'FETCH_USER_PROFILE';
export const SET_USER_PROFILE_DETAILS = 'SET_USER_PROFILE_DETAILS';
export const FETCH_USER_PROFILE_PICTURE = 'FETCH_USER_PROFILE_PICTURE';
export const SET_USER_PROFILE_PICTURE = 'SET_USER_PROFILE_PICTURE';
export const SET_ORG_CHART = 'SET_ORG_CHART';
export const FETCH_ORG_CHART = 'FETCH_ORG_CHART';
export const USER_PROFILE_RESPONSE_ERROR = 'USER_PROFILE_RESPONSE_ERROR';
export const ORG_CHART_RESPONSE_ERROR = 'ORG_CHART_RESPONSE_ERROR';
export const CLEAR_PROFILE_DATA = 'CLEAR_PROFILE_DATA';
export const UPDATE_USER_PROFILE_DATA = 'UPDATE_USER_PROFILE_DATA';
export const UPDATE_USER_PROFILE_REQUEST = 'UPDATE_USER_PROFILE_REQUEST';
export const UPDATE_USER_PROFILE_ERROR = 'UPDATE_USER_PROFILE_ERROR';
export const FETCH_MY_CONTACTS = 'FETCH_MY_CONTACTS';
export const SET_MY_CONTACTS = 'SET_MY_CONTACTS';
export const MY_CONTACTS_RESPONSE_ERROR = 'MY_CONTACTS_RESPONSE_ERROR';
export const REMOVE_SINGLE_CONTACT = 'REMOVE_SINGLE_CONTACT';
export const INITIATE_UPLOAD_PROFILE_PIC = 'INITIATE_UPLOAD_PROFILE_PIC';
export const UPLOAD_PROFILE_PIC_SUCCESS = 'UPLOAD_PROFILE_PIC_SUCCESS';
export const UPLOAD_PROFILE_PIC_ERROR = 'UPLOAD_PROFILE_PIC_ERROR';
export const CLEAR_USER_PROFILE_UPDATE_ERROR = 'CLEAR_USER_PROFILE_UPDATE_ERROR';
export const REMOVE_PROFILE_PIC = 'REMOVE_PROFILE_PIC';
export const COMPLETE_REMOVE_PROFILE_PIC = 'COMPLETE_REMOVE_PROFILE_PIC';

export interface UserProfileState {
  userProfileDetails: UserProfileModel;
  userProfilePicture: UserProfilePicture;
  userOrgChart: Users;
  profileDataError: boolean;
  chartDataError: boolean;
  isUserProfileUpdate: boolean;
  isFetchUserProfile: boolean;
  userProfileUpdateErrorMessage: string;
  isUserProfileUpdateError: boolean;
  userContactList: ContactList[];
  isFetchingMyContacts: boolean;
  fetchingMyContactsError: boolean;
  removeContactSucess: boolean;
  uploadSuccess: boolean;
  uploadError: boolean;
  isUploading: boolean;
  removeProfilePicSucess: boolean;
  isFetchingOrgChart: boolean;
}

export interface LoadUserProfileDetailsAction {
  type: typeof SET_USER_PROFILE_DETAILS;
  userProfile: UserProfileModel;
}

export interface LoadUserProfilePictureAction {
  type: typeof SET_USER_PROFILE_PICTURE;
  userProfilePicture: UserProfilePicture;
}

export interface OrgChartState {
  orgChartState: Users;
}

export interface LoadOrgChartAction {
  type: typeof SET_ORG_CHART;
  orgChartusers: Users;
  chartDataError: boolean;
  isFetchingOrgChart: boolean;
}

export interface UserProfileResponseError {
  type: typeof USER_PROFILE_RESPONSE_ERROR;
  profileDataError?: boolean;
}

export interface OrgChartResponseError {
  type: typeof ORG_CHART_RESPONSE_ERROR;
  chartDataError?: boolean;
}

export interface ClearUserProfile {
  type: typeof CLEAR_PROFILE_DATA;
}
export interface UpdateUserProfile {
  type: typeof UPDATE_USER_PROFILE_DATA;
  isUserProfileUpdate?: boolean;
  userProfileUpdateErrorMessage: string;
  isUserProfileUpdateError: boolean;
}
export interface UpdateUserProfileRequest {
  type: typeof UPDATE_USER_PROFILE_REQUEST;
  isUserProfileUpdate?: boolean;
  isUserProfileUpdateError?: boolean;
  userProfileUpdateErrorMessage?: string;
}

export interface UpdateUserProfileError {
  type: typeof UPDATE_USER_PROFILE_ERROR;
  isUserProfileUpdate?: boolean;
}

export interface FetchUserProfile {
  type: typeof FETCH_USER_PROFILE;
  isFetchUserProfile?: boolean;
  uploadSuccess?: boolean;
}

export interface FetchMyContacts {
  type: typeof FETCH_MY_CONTACTS;
  isFetchingMyContacts: boolean;
}

export interface SetMyContacts {
  type: typeof SET_MY_CONTACTS;
  userContactList: ContactList;
  isFetchingMyContacts: boolean;
  removeContactSucess: boolean;
}

export interface MyContactsResponseError {
  type: typeof MY_CONTACTS_RESPONSE_ERROR;
  isFetchingMyContacts: boolean;
  fetchingMyContactsError: boolean;
}

export interface RemoveSingleContactType {
  type: typeof REMOVE_SINGLE_CONTACT;
  removeContactSucess: boolean;
}

export interface InitiateUploadProfilePic {
  type: typeof INITIATE_UPLOAD_PROFILE_PIC;
  uploadSuccess: boolean;
  isUploading: boolean;
}

export interface UploadProfilePictureSuccessType {
  type: typeof UPLOAD_PROFILE_PIC_SUCCESS;
  uploadSuccess: boolean;
  isUploading: boolean;
}

export interface UploadProfilePictureErrorType {
  type: typeof UPLOAD_PROFILE_PIC_ERROR;
  uploadError: boolean;
}

export interface ClearUserProfileUpdateError {
  type: typeof CLEAR_USER_PROFILE_UPDATE_ERROR;
  isUserProfileUpdate?: boolean;
  isUserProfileUpdateError?: boolean;
  userProfileUpdateErrorMessage?: string;
}

export interface CompleteRemoveProfilePic {
  type: typeof COMPLETE_REMOVE_PROFILE_PIC;
  removeProfilePicSucess: boolean;
}

export interface RemoveUserProfilePic {
  type: typeof REMOVE_PROFILE_PIC;
  removeProfilePicSucess: boolean;
}

export interface FetchOrgChart {
  type: typeof FETCH_ORG_CHART;
  isFetchingOrgChart: boolean;
}
