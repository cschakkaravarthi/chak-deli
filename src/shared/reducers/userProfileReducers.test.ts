import userProfileReducers, { initialState } from './userProfileReducers';
import { SET_USER_PROFILE_DETAILS, SET_ORG_CHART, USER_PROFILE_RESPONSE_ERROR, ORG_CHART_RESPONSE_ERROR, CLEAR_PROFILE_DATA } from '../types/userProfileTypes';
import { dummyProfileInfo } from '../../shared/models/UserInfo.model';
import { dummyUsers } from '../../shared/models/OrgChart.model';

describe('userProfileReducers', () => {
  it('Handles default case', () => {
    const state = userProfileReducers(undefined, {});
    expect(state).toEqual(initialState);
  });

  it('Handles setProfileDetails action', () => {
    const action = { type: SET_USER_PROFILE_DETAILS, userProfileDetails: dummyProfileInfo };
    const state = userProfileReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles setUserOrgChart action', () => {
    const action = { type: SET_ORG_CHART, userOrgChart: dummyUsers };
    const state = userProfileReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles UserProfileResponseError action', () => {
    const action = { type: USER_PROFILE_RESPONSE_ERROR, profileDataError: true };
    const state = userProfileReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles OrgChartResponseError action', () => {
    const action = { type: ORG_CHART_RESPONSE_ERROR, profileDataError: true };
    const state = userProfileReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles clearProfileData action', () => {
    const action = { type: CLEAR_PROFILE_DATA, userProfileDetails: { title: '' }, userOrgChart: {}, profileDataError: false, chartDataError: false };
    const state = userProfileReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });
});
