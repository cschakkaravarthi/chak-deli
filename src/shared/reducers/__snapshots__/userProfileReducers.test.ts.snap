// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userProfileReducers Handles OrgChartResponseError action 1`] = `
Object {
  "chartDataError": true,
  "fetchingMyContactsError": false,
  "isFetchUserProfile": true,
  "isFetchingMyContacts": true,
  "isFetchingOrgChart": true,
  "isUploading": false,
  "isUserProfileUpdate": true,
  "isUserProfileUpdateError": false,
  "profileDataError": false,
  "removeContactSucess": false,
  "removeProfilePicSucess": false,
  "uploadError": false,
  "uploadSuccess": false,
  "userContactList": Object {},
  "userOrgChart": Object {},
  "userProfileDetails": Object {},
  "userProfilePicture": Object {},
  "userProfileUpdateErrorMessage": "",
}
`;

exports[`userProfileReducers Handles UserProfileResponseError action 1`] = `
Object {
  "chartDataError": false,
  "fetchingMyContactsError": false,
  "isFetchUserProfile": false,
  "isFetchingMyContacts": true,
  "isFetchingOrgChart": true,
  "isUploading": false,
  "isUserProfileUpdate": true,
  "isUserProfileUpdateError": false,
  "profileDataError": true,
  "removeContactSucess": false,
  "removeProfilePicSucess": false,
  "uploadError": false,
  "uploadSuccess": false,
  "userContactList": Object {},
  "userOrgChart": Object {},
  "userProfileDetails": Object {},
  "userProfilePicture": Object {},
  "userProfileUpdateErrorMessage": "",
}
`;

exports[`userProfileReducers Handles clearProfileData action 1`] = `
Object {
  "chartDataError": false,
  "fetchingMyContactsError": false,
  "isFetchUserProfile": true,
  "isFetchingMyContacts": true,
  "isFetchingOrgChart": true,
  "isUploading": false,
  "isUserProfileUpdate": true,
  "isUserProfileUpdateError": false,
  "profileDataError": false,
  "removeContactSucess": false,
  "removeProfilePicSucess": false,
  "uploadError": false,
  "uploadSuccess": false,
  "userContactList": Object {},
  "userOrgChart": Object {},
  "userProfileDetails": Object {
    "title": "",
  },
  "userProfilePicture": Object {},
  "userProfileUpdateErrorMessage": "",
}
`;

exports[`userProfileReducers Handles setProfileDetails action 1`] = `
Object {
  "chartDataError": false,
  "fetchingMyContactsError": false,
  "isFetchUserProfile": false,
  "isFetchingMyContacts": true,
  "isFetchingOrgChart": true,
  "isUploading": false,
  "isUserProfileUpdate": true,
  "isUserProfileUpdateError": false,
  "profileDataError": false,
  "removeContactSucess": false,
  "removeProfilePicSucess": false,
  "uploadError": false,
  "uploadSuccess": false,
  "userContactList": Object {},
  "userOrgChart": Object {},
  "userProfileDetails": undefined,
  "userProfilePicture": Object {},
  "userProfileUpdateErrorMessage": "",
}
`;

exports[`userProfileReducers Handles setUserOrgChart action 1`] = `
Object {
  "chartDataError": undefined,
  "fetchingMyContactsError": false,
  "isFetchUserProfile": true,
  "isFetchingMyContacts": true,
  "isFetchingOrgChart": undefined,
  "isUploading": false,
  "isUserProfileUpdate": true,
  "isUserProfileUpdateError": false,
  "profileDataError": false,
  "removeContactSucess": false,
  "removeProfilePicSucess": false,
  "uploadError": false,
  "uploadSuccess": false,
  "userContactList": Object {},
  "userOrgChart": undefined,
  "userProfileDetails": Object {},
  "userProfilePicture": Object {},
  "userProfileUpdateErrorMessage": "",
}
`;
